crates/hir-def/src/body/lower.rs

```diff
@@ -1598,6 +1598,10 @@ impl ExprCollector<'_> {
                 for (id, _) in current_is_used.into_iter() {
                     binding_list.check_is_used(self, id);
                 }
+                if pats.len() == 1 {
+                    // Leading pipe without real OR pattern. Leaving an one-item OR pattern may confuse later stages.
+                    return pats.into_iter().next().unwrap();
+                }
```

Comment:
```suggestion
                if let &[pat] = &*pats {
                    // Leading pipe without real OR pattern. Leaving an one-item OR pattern may confuse later stages.
                    return pat;
                }
```

Reply:
But then I only get a reference and `Pat` is expensive to clone...

Reply:
But `pat `here is a `PatId` no? (the code here copies, not clones)

Reply:
Oh yeah updated.